From 505c423432e45dc6965321a4c2cd03cabccdacec Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Tue, 9 Mar 2021 12:36:37 -0800
Subject: [PATCH 1766/1852] drm: xlnx: mixer: Add port OF node to DRM CRTC

commit 769a19920173b4dfe003ccc743034f2a53ed04cc from
https://github.com/Xilinx/linux-xlnx.git

Currently the DRM CRTC initialized by the mixer driver does not fill out
the port OF node it is related to. This is required by DRM encoders
because it needs to have a list of possible CRTCs incase there are
multiple CRTCs connected to a single encoder.

So this patch attaches the port OF node to the DRM CRTC during DTS
parsing.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Signed-off-by: Jianqiang Chen <jianqiang.chen@xilinx.com>
Acked-by: Varunkumar Allagadapa <varunkumar.allagadapa@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_mixer.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_mixer.c b/drivers/gpu/drm/xlnx/xlnx_mixer.c
index 826217b74664..b4471c1fa0b3 100644
--- a/drivers/gpu/drm/xlnx/xlnx_mixer.c
+++ b/drivers/gpu/drm/xlnx/xlnx_mixer.c
@@ -2236,7 +2236,7 @@ static int xlnx_mix_dt_parse(struct device *dev, struct xlnx_mix *mixer)
 {
 	struct xlnx_mix_plane *planes;
 	struct xlnx_mix_hw *mixer_hw;
-	struct device_node *node, *vtc_node;
+	struct device_node *node, *vtc_node, *port;
 	struct xlnx_mix_layer_data *l_data;
 	struct resource	res;
 	int ret, l_cnt, i;
@@ -2347,6 +2347,14 @@ static int xlnx_mix_dt_parse(struct device *dev, struct xlnx_mix *mixer)
 		ret = xlnx_mix_parse_dt_logo_data(node, mixer_hw);
 		return ret;
 	}
+
+	/* Fill out crtc port OF node */
+	for_each_child_of_node(node, port) {
+		if (!port->name || of_node_cmp(port->name, "port"))
+			continue;
+		mixer->crtc.crtc.port = port;
+		break;
+	}
 	return 0;
 }
 
-- 
2.31.1

